packages/ui-link/src/Link/styles.ts

```diff
@@ -142,57 +200,56 @@ const generateStyle = (
     margin: 0,
     padding: 0,
     textAlign: 'inherit',
-    ...(variant ? variantStyles[variant] : { fontSize: '1em' })
+    fontSize: variantStyles.fontSize,
+    lineHeight: variantStyles.lineHeight,
+    ...(variantStyles.textDecoration && {
+      textDecoration: variantStyles.textDecoration
+    })
   }
 
   const inverseStyles = {
-    color: componentTheme.colorInverse,
-    '&:focus': {
-      outlineColor: componentTheme.focusInverseIconOutlineColor
-    },
-    '&:hover, &:focus, &:active': {
-      color: componentTheme.colorInverse
-    }
-  }
-  const variantIconStyles = {
-    inline: {
-      paddingInlineStart:
-        iconPlacement === 'start' ? 0 : componentTheme.iconPlusTextMargin,
-      paddingInlineEnd:
-        iconPlacement === 'start' ? componentTheme.iconPlusTextMargin : 0
-    },
-    'inline-small': {
-      paddingInlineStart:
-        iconPlacement === 'start' ? 0 : componentTheme.iconPlusTextMarginSmall,
-      paddingInlineEnd:
-        iconPlacement === 'start' ? componentTheme.iconPlusTextMarginSmall : 0
+    color: componentTheme.onColorTextColor,
+    // This needs stronger specificity than `View`
+    '&&&&&:focus': {
+      outlineColor: sharedTokens.focusOutline.onColor
```

Comment:
This is ugly :( As I see we're now doing double focus ring calculations, first here with the util and in `View`. Lets leave it as is now, when `View` is refactored we should add a prop to not show any focus ring properly. (as I see from View's code `withFocusOutline` is just partly reming the focus ring CSS classes) @balzss